### PR TITLE
Change location of .vegastrike folder on Windows - 0.8.x

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [X] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues
- Implements #550 for 0.8.x

Purpose:
- What is this pull request trying to do? Port #551 to 0.8.x, and then change location of .vegastrike and .pu folders on Windows, from "My Documents" to `C:\Users\<user>\AppData\Local`
- What release is this for? 0.8.x; 0.9.x pending
- Is there a project or milestone we should apply this to? 0.8.x; 0.9.x pending

### CODE REVIEW

**Comment by Loki1950** [comment]
BTW that is the location that 0.05 placed the config file so there should be #Define with the relevant code still about.

**Review by BenjamenMeyer — COMMENTED** [review]
LGTM, but one question: does it preserve the legacy location?

**Comment by stephengtuggy** [comment]
> BTW that is the location that 0.05 placed the config file so there should be #Define with the relevant code still about.

Really? I thought I remembered 0.5.x trying to store all the data under Program Files.

Sure enough, looks like you're right. The My Documents location was introduced by @pheonixstorm in https://github.com/vegastrike/Vega-Strike-Engine-Source/commit/0d8aaddf719ab107120122d1b411043cafc233e9 , in 2012. However, did that make it into 0.5.1r1, the last release from SVN?

**Comment by stephengtuggy** [comment]
I see that 0.5.1.r1 was released April 1, 2012: https://forums.vega-strike.org/viewtopic.php?f=0&t=18088 https://github.com/vegastrike/Vega-Strike-Engine-Source/commit/0d8aaddf719ab107120122d1b411043cafc233e9 was committed July 22, 2012. So I don't think that change made it into any of the releases until we switched over to GitHub, at which point VS was no longer playable on Windows anyway.

**Comment by stephengtuggy** [comment]
The commit before that, https://github.com/vegastrike/Vega-Strike-Engine-Source/commit/53de7524648fe43a8093490a86ab2f874986bfd2 , dated April 16, 2012, used `getenv("USERPROFILE")` . And the commit before that is the one with the behavior I remember: https://github.com/vegastrike/Vega-Strike-Engine-Source/commit/97fb92cea69c1eb74694122515339a54dc043888 , dated October 1, 2003.

**Review by Loki1950 — APPROVED** [review]

File: engine/src/vs_globals.h [hunk 1]
```diff
@@ -72,41 +98,41 @@ extern Mission    *mission;
 template < class MyType >
 class LeakVector
 {
-    std::vector< MyType > *active_missions;
+    std::vector< MyType > *internal_vector_p_;
```
Comment (BenjamenMeyer) [inline engine/src/vs_globals.h:101]: for clarity, this should be declared private (default for class) or protected.

Comment (stephengtuggy) [reply engine/src/vs_globals.h:102]: Done.

**Review by BenjamenMeyer — APPROVED** [review]

**Pull request merged**